azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -922,12 +933,23 @@ func flattenAzureRmVirtualMachineScaleSetStorageProfileDataDisk(disks *[]compute
 	return result
 }
 
-func flattenAzureRmVirtualMachineScaleSetStorageProfileImageReference(profile *compute.ImageReference) []interface{} {
+func flattenAzureRmVirtualMachineScaleSetStorageProfileImageReference(image *compute.ImageReference) []interface{} {
 	result := make(map[string]interface{})
-	result["publisher"] = *profile.Publisher
-	result["offer"] = *profile.Offer
-	result["sku"] = *profile.Sku
-	result["version"] = *profile.Version
+	if image.Publisher != nil {
+		result["publisher"] = *image.Publisher
+	}
+	if image.Offer != nil {
+		result["offer"] = *image.Offer
+	}
+	if image.Sku != nil {
+		result["sku"] = *image.Sku
+	}
+	if image.Version != nil {
+		result["version"] = *image.Version
+	}
+	if image.ID != nil {
+		result["id"] = *image.ID
+	}
```

Comment:
Hey @tombuildsstuff thanks for a quick turnaround on this; I provisioned VMSS using TF v. 0.9.11, then ran `terraform plan` using my branch. Got the message 

No changes. Infrastructure is up-to-date.

I believe this is expected behavior.